airbyte-cdk/python/airbyte_cdk/sources/declarative/interpolation/jinja.py

```diff
@@ -75,6 +91,11 @@ def _literal_eval(self, result):
 
     def _eval(self, s: str, context):
         try:
+            ast = self._environment.parse(s)
+            undeclared = meta.find_undeclared_variables(ast)
+            undeclared_not_in_context = {var for var in undeclared if var not in context}
+            if undeclared_not_in_context:
+                raise ValueError(f"Jinja macro has undeclared variables: {undeclared}. Context: {context}")
```

Comment:
this should probably be`undeclared_not_in_context` otherwise we'll list valid template variables that are in the context